ArdusatSDK.cpp

```diff
@@ -1183,14 +1404,15 @@ boolean Temperature::initialize(void) {
  * @brief   Takes a reading from the sensor
  * @ingroup temperature
  */
-void Temperature::read(void) {
-  this->header.timestamp = millis();
-
+boolean Temperature::readSensor(void) {
   if (this->header.sensor_id == SENSORID_TMP102) {
```

Comment:
same comment as for the RGBLight - define a default version, which is the TMP driver, and then override this function for the `TemperatureMLX` class.